src/FileType_HDF5.f90

```diff
@@ -1153,6 +1162,109 @@ FUNCTION pathexists_HDF5FileType(thisHDF5File,path) RESULT(bool)
     ENDFUNCTION pathexists_HDF5FileType
 !
 !-------------------------------------------------------------------------------
+!> @brief   Returns whether an HDF5 File or data set has compression
+!> @param   thisHDF5File the HDF5FileType object to interrogate
+!> @param   path (optional) the path to a dataset in the file
+!> @returns bool the logical result of whether the specified object has
+!>          compression.
+!>
```

Comment:
Maybe just clarify a bit more either in the header or the `IF(PRESENT())` check for the path that it will return the file's `%hascompression`.